ath79: remove wmac mtd-mac-addres for UniFi AC family

rogerpueyo opened the pull request:
~~The Ubiquiti Unifi AC Mesh devices have the following MAC addresses when running the stock firmware:~~
~~· eth0: XX:XX:XX:X0:XX:XX~~
~~· ath0: XX:XX:XX:X1:XX:XX (2.4 GHz)~~
~~· ath1: XX:XX:XX:X2:XX:XX (5 GHz)~~

~~The commit fixes the MAC address for the 2.4 GHz radio, which got the same MAC as eth0.~~

~~MAC addresses assignment for the Unifi AC Lite devices, which have a commonbase .dtsi file, is kept as it was.~~

The Ubiquiti UniFi AC family of devices had an incorrect mtd-mac-address for wmac (the WiSoC's integrated 2.4 GHz radio) which forced it to have the same MAC as eth0. This commit removes it, so that MAC assignment is coherent with the stock firmware and the ar71xx target.

adrianschmutzler commented:
Hi,

Looks like qca9563_ubnt_unifiac.dtsi is also sourced by unifiac-pro:
```
adsc@buildfff:/data/openwrt$ grep -rni "ubnt_unifiac.dtsi" target/linux/ath79/
target/linux/ath79/dts/qca9563_ubnt_unifiac-pro.dtsi:3:#include "qca9563_ubnt_unifiac.dtsi"
target/linux/ath79/dts/qca9563_ubnt_unifiac-lite.dtsi:3:#include "qca9563_ubnt_unifiac.dtsi"
```

So, you would have to add wmac node there.

However, I would favor another approach:
`mtd-mac-address = <&eeprom 0x1002>;` is actually redundant, as `mtd-cal-data = <&eeprom 0x1000>;` will use that address anyway. So, you could just keep the wmac block in qca9563_ubnt_unifiac.dtsi, just without the mtd-mac-address:

```
&wmac {
	status = "okay";

	mtd-cal-data = <&eeprom 0x1000>;
};
```

This will then already set up AC Mesh correctly, for the other affected devices you could just add

```
&wmac {
	mtd-mac-address = <&eeprom 0x0>;
};
```

into the relevant DTS/DTSI files.

Despite all that, I wonder that mtd-mac-address is needed at all with Ubiquiti...

adrianschmutzler commented:
From reading your recent commit message, it looks like you've only tested AC-Mesh, and keep the other devices to 0x0 just because they had this setting before? I'm just asking because I suspect that the 0x0 will prove wrong also for some of the others...
If you have tested others that actually had 0x0, I would be glad if you added this to the commit message.

adrianschmutzler commented:
The current version is what I had in mind. :-)

Nitpick: I would use "wmac MAC address" instead of "wireless MAC address" in the commit title, as the AC Mesh has two wireless interface, and you are only touching one (as you correctly specify in the commit _message_).

rogerpueyo commented:
Hi Adrian,

> just without the mtd-mac-address:
> 
> ```
> &wmac {
> 	status = "okay";
> 
> 	mtd-cal-data = <&eeprom 0x1000>;
> };
> ```

Yes, that's enough, the MAC address is extracted correctly from the caldata.
> This will then already set up AC M



> From reading your recent commit message, it looks like you've only tested AC-Mesh, and keep the other devices to 0x0 just because they had this setting before? I'm just asking because I suspect that the 0x0 will prove wrong also for some of the others...
> If you have tested others that actually had 0x0, I would be glad if you added this to the commit message.

Unfortunately, I only have a UniFi AC Mesh, so I can't test the other 4 devices. Right now they're getting the same MAC for eth0 and the wmac, and I'm pretty sure this is not what the stock firmware does. But, since I can't check it... :(



> Nitpick: I would use "wmac MAC address" instead of "wireless MAC address" in the commit title, as the AC Mesh has two wireless interface, and you are only touching one (as you correctly specify in the commit _message_).

Totally, changed that on the last force-pushed version :)

adschm reviewed (approved):
Looks good and is complete.

If someone can test on Lite/LR/Mesh-Pro/Pro, please do so. If we find out that other devices are wrong, too, changing them should now be easier (as one can just remove the wmac node).

adrianschmutzler commented:
@rotanid Can you check what's the MAC address of wmac (2.4 GHz) on your Ubiquiti UniFi AC LR?
Is it the same as eth0 (&art 0x0) or should it be set by caldata (&art 0x1002) as for the AC Mesh?

@blocktrron FYI

rogerpueyo commented:
Hi @achterin 

We just found that the UniFi AC Mesh devices had a wrong MAC address assignation for wmac (2.4 GHz radio) compared to the stock firmware. It is likely that the Unifi AC-Mesh Pro (for which you added support as per commit:e420b7b5c517d03e1282b5889065e22143f9c4ef) also has the same problem. I wonder if you could possibly check if the stock Ubiquiti firmware uses the same MAC addresses on eth0 and the 2.4 GHz radio.

Thanks

adrianschmutzler commented:
@rogerpueyo 

Just remembered the obvious and had a look into ar71xx:
https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/files/arch/mips/ath79/mach-ubnt-unifiac.c

There, wmac is set up without setting the MAC address (ath79_register_wmac without second argument) for Lite/Mesh/Pro/Mesh-Pro.

Putting together the actual test and this information, I vote for removing 0x0 from ALL devices in the commit (with exactly this justification). But since the is no need to hurry, we can wait for at least one additional test.

rotanid commented:
> @rotanid Can you check what's the MAC address of wmac (2.4 GHz) on your Ubiquiti UniFi AC LR _with stock firmware_?
> Is it the same as eth0 (&art 0x0) or should it be set by caldata (&art 0x1002) as for the AC Mesh?

please send me more detailed information on what you need via mail, thanks!

rogerpueyo commented:
> @rogerpueyo
> 
> Just remembered the obvious and had a look into ar71xx:
> https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/files/arch/mips/ath79/mach-ubnt-unifiac.c
> 
> There, wmac is set up without setting the MAC address (ath79_register_wmac without second argument) for Lite/Mesh/Pro/Mesh-Pro.
> 
> Putting together the actual test and this information, I vote for removing 0x0 from ALL devices in the commit (with exactly this justification). But since there is no need to hurry, we can wait for at least one additional test.

adrianschmutzler commented:
I will an e-mail to @rotanid

rogerpueyo commented:
> I will an e-mail to @rotanid

Did that already :)

Sorry, I closed the PR by mistake. Clumsy hand on the mouse.

adrianschmutzler commented:
Ah, so now he will have two explanation mails :-)

rotanid commented:
i just checked the AC-LR, eth0 and ath0 differ and are at 0x0 and 0x1002 in EEPROM partition.
Adrian got a longer answer via mail.

adrianschmutzler commented:
@rogerpueyo So, AC-LR matches the behavior of AC Mesh, and the setup used for all UniFi AC devices on ar71xx.
I'm convinced now that the best solution will be to just remove the mtd-mac-address for all Unifi AC devices (yielding a one-line patch...).

rogerpueyo commented:
> @rogerpueyo So, AC-LR matches the behavior of AC Mesh, and the setup used for all UniFi AC devices on ar71xx.
> I'm convinced now that the best solution will be to just remove the mtd-mac-address for all Unifi AC devices (yielding a one-line patch...).

Absolutely. Just pushed. :)

adrianschmutzler commented:
Sorry for stirring this again and again, but I would try to preserve as much of our/your findings in the commit message. Thus, maybe add back the MAC address scheme
· eth0: XX:XX:XX:X0:XX:XX
· ath0: XX:XX:XX:X1:XX:XX (2.4 GHz)
· ath1: XX:XX:XX:X2:XX:XX (5 GHz)
and state which devices we actually tested on.

If you like, you can also add the empty line again, I liked it.

adschm reviewed (approved):
Fine.

achterin commented:
will test mesh-pro this weekend

most23 commented:
Tested on Unifi AC Lite:
Before patch:
eth0: XX:XX:XX:F2:XX:XX
ath0: XX:XX:XX:F2:XX:XX
ath1: XX:XX:XX:F4:XX:XX
After patch:
eth0: XX:XX:XX:F2:XX:XX
ath0: XX:XX:XX:F3:XX:XX
ath1: XX:XX:XX:F4:XX:XX
Do not know if this corresponds to stock or not.

adrianschmutzler commented:
@most23 Thanks for the data, but actually the most interesting thing would be the stock setup on AC Lite.

most23 commented:
@adrianschmutzler I have now loaded stock firmware (4.0.54.10625) on AC Lite and can confirm patch corresponds to stock setup (F2-F3-F4 pattern).

rogerpueyo commented:
> @adrianschmutzler I have now loaded stock firmware (4.0.54.10625) on AC Lite and can confirm patch corresponds to stock setup (F2-F3-F4 pattern).

Cool, thanks @most23 !

I've rebased master and updated the commit message.

adrianschmutzler commented:
Your From: and Signed-off-by: do not match.

If you want to set From: to Signed-off-by:, you can just use `git commit --am --reset-author`

adrianschmutzler commented:
And you are missing the second "s" in mtd-mac-address in the commit title.

rogerpueyo committed:
ath79: remove wmac mtd-mac-address for UniFi AC family

The MAC address for the wmac 2.4 GHz radio of the Ubiquiti UniFi AC
family of devices is actually embedded in the mtd-cal-data, so there
is no need for mtd-mac-address (which was incorrectly forcing wmac
to have the same MAC as eth0). This makes it coherent with the stock
firmware and the ar71xx target:

 · XX:XX:XX:X0:XX:XX eth0
 · XX:XX:XX:X1:XX:XX ath0/wlan1 (2.4 GHz)
 · XX:XX:XX:X2:XX:XX ath1/wlan0 (5 GHz)

Checked on a UniFi AC Mesh, a UniFi AC LR and a UniFi Lite.

Signed-off-by: Roger Pueyo Centelles <roger.pueyo@guifi.net>

adrianschmutzler commented:
Thanks, I've pulled this into my staging tree and will merge it shortly.
